src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java

```diff
@@ -81,28 +76,41 @@ public void create(EntityDescriptor descriptor,
 
     @Override
     protected void onBeforeSaveChecks(Object entity) {
-        getMultiLanguageString(entity).data().forEach((language, text) -> {
+        MultiLanguageString multiLanguageString = getMultiLanguageString(entity);
+        multiLanguageString.data().forEach((language, text) -> {
             if (Strings.areEqual(language, MultiLanguageString.FALLBACK_KEY)) {
                 return;
             }
-            if (supportedLanguages != null && !supportedLanguages.contains(language)) {
+            if (multiLanguageString.getValidLanguages() != null && !multiLanguageString.getValidLanguages().contains(language)) {
                 throw Exceptions.createHandled()
-                                .withNLSKey("MultiLanguageString.invalidLanguage")
-                                .set("language", language)
-                                .set("text", text)
-                                .set("field", getField().getName())
-                                .handle();
+                        .withNLSKey("MultiLanguageString.invalidLanguage")
+                        .set("language", language)
+                        .set("text", text)
+                        .set("field", getField().getName())
+                        .handle();
             }
         });
 
         super.onBeforeSaveChecks(entity);
     }
 
-    @SuppressWarnings("unchecked")
-    protected MultiLanguageString getMultiLanguageString(Object entity) {
-        MultiLanguageString multiLanguageString = new MultiLanguageString();
-        multiLanguageString.setData((Map<String, String>) super.getValueFromField(entity));
-        return multiLanguageString;
+    /**
+     * Bypasses {@link BaseMapProperty#getValueFromField(Object)} to access the actual MultiLanguageString entity.
+     *
+     * @see Property#getValueFromField(Object)
+     */
+    protected MultiLanguageString getMultiLanguageString(Object target) {
+        try {
+            return (MultiLanguageString) field.get(target);
```

Comment:
is this guaranteed to work with composites and mixins - rather use getValueFromField which handles all this correnctly?

Reply:
MultiLanguageString is a BaseMapProperty, so its getValueFromField returns only the data as a Map. We need access to the actual object here though.
So this method basically copies getValueFromField() from Property (parent of BaseMapProperty), but returns a MultiLanguageString instead of Object.

Overwriting getValueFromField() with that functionality results in errors during other parts of the pipeline. This keeps required changes minimal.

Reply:
then the accessPath has to be manually applied at the correct places - see BaseEntityRefProperty. Add a test within a composite and within a Mixin to reproduce the error and ensure that it works properly :)